handler_test.go

```diff
@@ -0,0 +1,1982 @@
+							},
+							ResolutionFailures: 0,
+							Conditions: []metav1.Condition{
+								{
+									Type:    ConditionDegraded,
+									Status:  metav1.ConditionFalse,
+									Reason:  dns.RcodeToString[dns.RcodeSuccess],
+									Message: rcodeMessage[dns.RcodeSuccess],
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		dnsQueryTestCases: []test.Case{
+			{
+				Qname: "www.example.com.",
+				Qtype: dns.TypeA,
+				Rcode: dns.RcodeNameError,
+			},
+		},
+		expectedStatuses: []ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+			{
+				ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+					{
+						DNSName: "www.example.com.",
+						ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+							{
+								IP:         "1.1.1.1",
+								TTLSeconds: 30,
+							},
+							{
+								IP:         "1.1.1.2",
+								TTLSeconds: 30,
+							},
+						},
+						ResolutionFailures: 1,
+						Conditions: []metav1.Condition{
+							{
+								Type:    ConditionDegraded,
+								Status:  metav1.ConditionTrue,
+								Reason:  dns.RcodeToString[dns.RcodeNameError],
+								Message: rcodeMessage[dns.RcodeNameError],
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		name: "Increment resolutionFailure field in regular dns name resolver object's status and update ttlSeconds and lastLookupTime fields",
+		dnsNameResolvers: []ocpnetworkapiv1alpha1.DNSNameResolver{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "regular",
+					Namespace: "dns",
+				},
+				Spec: ocpnetworkapiv1alpha1.DNSNameResolverSpec{
+					Name: "www.example.com.",
+				},
+				Status: ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+					ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+						{
+							DNSName: "www.example.com.",
+							ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+								{
+									IP:         "1.1.1.1",
+									TTLSeconds: 2,
+								},
+								{
+									IP:         "1.1.1.2",
+									TTLSeconds: 2,
+								},
+							},
+							ResolutionFailures: 1,
+							Conditions: []metav1.Condition{
+								{
+									Type:    ConditionDegraded,
+									Status:  metav1.ConditionTrue,
+									Reason:  dns.RcodeToString[dns.RcodeNameError],
+									Message: rcodeMessage[dns.RcodeNameError],
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		dnsQueryTestCases: []test.Case{
+			{
+				Qname: "www.example.com.",
+				Qtype: dns.TypeA,
+				Rcode: dns.RcodeNameError,
+			},
+		},
+		expectedStatuses: []ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+			{
+				ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+					{
+
+						DNSName: "www.example.com.",
+						ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+							{
+								IP:         "1.1.1.1",
+								TTLSeconds: 5,
+							},
+							{
+								IP:         "1.1.1.2",
+								TTLSeconds: 5,
+							},
+						},
```

Comment:
When the DNS lookup is unsuccessful and if the next lookup time is within 5 seconds of the current time, then to avoid an immediate lookup at the TTL expiration, the corresponding `TTLSeconds` is updated to 5. This will delay the possible lookup failure by roughly 5 seonds, if at all.

Ok, I'll add a comment explaining the change in the TTL value.